### PR TITLE
Next dockerized

### PR DESCRIPTION
**Pull request by nstankov-bg**
Containerizing stuff.

### CODE REVIEW

File: docker/entrypoint/docker-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
+
+if [ ! -e node_modules ]; then
+#Running the installer, hold on to your butts.
+echo "---"
+echo "---"
+echo "#Running the installer, hold on to your butts."
+echo "---"
+echo "---"
+npm install
```
Comment (Dauta) [inline docker/entrypoint/docker-entrypoint.sh:10]: can we use `yarn install` here instead? generally it's not a good idea to mix npm and yarn. I think alpine-node should have yarn pre-installed as well.

File: docker-compose-osx.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    working_dir: /home/node/app
+    environment:
+      - NODE_ENV=development
+    volumes:
+      - codebase-docker-sync:/home/node/app
+      - codebase-docker-sync:/home/node/app/book
+    ports:
+      - "5000:5000"
+
+volumes:
```
Comment (Dauta) [inline docker-compose-osx.yml:15]: out of curiosity, why is there an osx-specific docker compose file?

File: docker/entrypoint/docker-entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
+
+if [ ! -e node_modules ]; then
+#Running the installer, hold on to your butts.
+echo "---"
+echo "---"
+echo "#Running the installer, hold on to your butts."
```
Comment (Dauta) [inline docker/entrypoint/docker-entrypoint.sh:7]: Optional: Not sure about this message 😅

**Review by Dauta — COMMENTED** [review]
Looks good overall. Left some comments. 
Mostly non-blocking things:
- why we need several different docker-compose files. 
- can we keep using `yarn` over `npm`. So we don't have to remove `yarn.lock` and can revert `package.lock`. Generally, it's not a good idea to keep both.
- can we rename `www` folder to something like `src`? I feel like `www` is from an entirely different ecosystem (might be wrong tho). The `src` file inside of the project can be renamed to something else as well, or even we can flatten the file structure and move everything from `src` to root. Nextjs is ok with that.

File: docker-compose-osx.yml (hunk 2 above)
Comment (nstankov-bg) [reply docker-compose-osx.yml:15]: Hey man, it's used to fix the sync issues for small files on OSX.
https://docker-sync.readthedocs.io/en/latest/

**Pull request merged**